great_expectations/execution_engine/split_and_sample/sparkdf_data_sampler.py

```diff
@@ -0,0 +1,119 @@
+import hashlib
+import logging
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.id_dict import BatchSpec
+from great_expectations.execution_engine.split_and_sample.data_sampler import (
+    DataSampler,
+)
+
+logger = logging.getLogger(__name__)
+
+try:
+    import pyspark
+    import pyspark.sql.functions as F
+
+    # noinspection SpellCheckingInspection
+    import pyspark.sql.types as sparktypes
+    from pyspark.sql import DataFrame
+
+except ImportError:
+    pyspark = None
+    DataFrame = None
+    F = None
+    # noinspection SpellCheckingInspection
+    sparktypes = None
+
+    logger.debug(
+        "Unable to load pyspark; install optional spark dependency for support."
+    )
+
+
+class SparkDataSampler(DataSampler):
+    """Methods for sampling a Spark dataframe."""
+
+    def sample_using_limit(self, df: DataFrame, batch_spec: BatchSpec) -> DataFrame:
+        """Sample the first n rows of data.
+
+        Args:
+            df: Spark dataframe.
+            batch_spec: BatchSpec with sampling_kwargs for limit sampling e.g. sampling_kwargs={"n": 100}.
+
+        Returns:
+            Pandas dataframe reduced to number of items specified in BatchSpec sampling kwargs.
+        """
+        self.verify_batch_spec_sampling_kwargs_exists(batch_spec)
+        self.verify_batch_spec_sampling_kwargs_key_exists("n", batch_spec)
+
+        n: int = batch_spec["sampling_kwargs"]["n"]
+        return df.limit(n)
+
+    @staticmethod
+    def sample_using_random(df: DataFrame, p: float = 0.1, seed: int = 1):
```

Comment:
The methods from here down are just moved from the execution engine. I can clean them up in a follow-on PR.